drivers/dma/dma_stm32.c

```diff
@@ -516,23 +641,40 @@ DMA_STM32_EXPORT_API int dma_stm32_reload(const struct device *dev, uint32_t id,
 
 	switch (stream->direction) {
 	case MEMORY_TO_PERIPHERAL:
+#if defined(CONFIG_DMA_STM32_V3)
+		LL_DMA_ConfigAddresses(dma, dma_stm32_id_to_stream(id), src, dst);
+#else
 		LL_DMA_SetMemoryAddress(dma, dma_stm32_id_to_stream(id), src);
 		LL_DMA_SetPeriphAddress(dma, dma_stm32_id_to_stream(id), dst);
+#endif /* CONFIG_DMA_STM32_V3 */
 		break;
 	case MEMORY_TO_MEMORY:
 	case PERIPHERAL_TO_MEMORY:
+#if defined(CONFIG_DMA_STM32_V3)
+		LL_DMA_ConfigAddresses(dma, dma_stm32_id_to_stream(id), dst, src);
+#else
 		LL_DMA_SetPeriphAddress(dma, dma_stm32_id_to_stream(id), src);
 		LL_DMA_SetMemoryAddress(dma, dma_stm32_id_to_stream(id), dst);
+#endif /* CONFIG_DMA_STM32_V3 */
```

Comment:
To maintain code readability, please add new `dma_stm32_foo` functions to abstract these LL API differences.

Reply:
adding `stm32_dma_set_mem_periph_address()` and `stm32_dma_set_periph_mem_address()`
to abstract the LL driver